### PR TITLE
Replaced showChooseDialog with MessageDialog class

### LINKED ISSUE #861: IntelliJ deprecated 'showChooseDialog' and there is no other UI element to replace it
IntelliJ deprecated 'showChooseDialog' and there is no other UI element to replace it.

Messages.Java class
image

We are using this choose dialog UI component for listing projects, if there is more than one project in the dashboard. 

image

### PR DESCRIPTION
**Pull request by anusreelakshmi934**
Fixes #861

### CODE REVIEW

File: src/main/java/io/openliberty/tools/intellij/actions/LibertyProjectAction.java [hunk 1]
```diff
@@ -129,6 +132,15 @@ protected final String[] buildFileToProjectNames(@NotNull List<BuildFile> list)
         return projectNames;
     }
 
+    protected final String[] buildFileToProjectPaths(@NotNull List<BuildFile> list) {
```
Comment (aparnamichael) [inline src/main/java/io/openliberty/tools/intellij/actions/LibertyProjectAction.java:135]: Can we write a single method which returns array of both projectname and project path by using HashMap?

**Comment by aparnamichael** [comment]
@anusreelakshmi934 Update the year in the file header.

File: src/main/java/io/openliberty/tools/intellij/actions/LibertyProjectAction.java (hunk 1 above)
Comment (anusreelakshmi934) [reply src/main/java/io/openliberty/tools/intellij/actions/LibertyProjectAction.java]: Yes. Made the change. @aparnamichael

File: src/main/java/io/openliberty/tools/intellij/actions/LibertyProjectAction.java [hunk 2]
```diff
@@ -46,13 +44,26 @@ public void actionPerformed(@NotNull AnActionEvent e) {
...
         if (!buildFileList.isEmpty()) {
-            final String[] projectNames = buildFileToProjectNames(buildFileList);
-            final int ret = Messages.showChooseDialog(project,
+            final List<Map<String, String>> projectDetails = buildFilesToProjectDetails(buildFileList);
+            final String[] projectNames = new String[projectDetails.size()];
+            final String[] projectPaths = new String[projectDetails.size()];
+
+            for (int i = 0; i < projectDetails.size(); i++) {
+                Map<String, String> details = projectDetails.get(i);
+                projectNames[i] = details.get("projectName");
```
Comment (aparnamichael) [inline src/main/java/io/openliberty/tools/intellij/actions/LibertyProjectAction.java:53]: can we avoid using this indexes and please check whether any Java methods available to get the **projectName** in one array and **projectPath** in another array.

Comment (anusreelakshmi934) [reply src/main/java/io/openliberty/tools/intellij/actions/LibertyProjectAction.java]: Changed to use Java streams

File: src/main/java/io/openliberty/tools/intellij/actions/LibertyProjectAction.java [hunk 3]
```diff
@@ -46,13 +45,19 @@ public void actionPerformed(@NotNull AnActionEvent e) {
         }
         List<BuildFile> buildFileList = getBuildFileList(project);
         if (!buildFileList.isEmpty()) {
-            final String[] projectNames = buildFileToProjectNames(buildFileList);
-            final int ret = Messages.showChooseDialog(project,
+            String[] projectName = extractBuildFileAttributes(buildFileList, BuildFile::getProjectName);
+            String[] projectPath = extractBuildFileAttributes(buildFileList, buildFile -> buildFile.getBuildFile().getPath());
```
Comment (mrglavas) [inline src/main/java/io/openliberty/tools/intellij/actions/LibertyProjectAction.java:49]: I'm not suggesting any changes here but noticed this is looping over the same list twice. Seems like we're gathering data into parallel arrays that could have been aggregated. Again, not suggesting you change this. Just something to think about when performance matters.

**Review by mrglavas — APPROVED** [review]

**Review by aparnamichael — APPROVED** [review]
Looks good!!

**Comment by mrglavas** [comment]
Discussed the path forward with @anusreelakshmi934 today. The plan is to revert the previous commit and I will then approve the PR again.

**Review by aparnamichael — APPROVED** [review]

**Review by mrglavas — APPROVED** [review]
Looks good. Thanks!

**Pull request merged**